telefunc/node/server/shield/codegen/generateShield.ts

```diff
@@ -0,0 +1,379 @@
+  })
+}
+
+function getImportPath(importer: string, importedFile: string) {
+  let importPath = path.relative(path.dirname(importer), importedFile)
+  importPath = toImport(importPath)
+  return importPath
+}
+
+function getTelefuncFileImportPath(telefuncFilePath: string) {
+  let importPath = path.basename(telefuncFilePath)
+  importPath = toImport(importPath)
+  return importPath
+}
+
+function toImport(importPath: string) {
+  assert(importPath.endsWith('.ts'))
+  importPath = importPath.slice(0, -1 * '.ts'.length)
+  if (process.platform === 'win32') {
+    importPath = importPath.split('\\').join('/')
+  }
+  return `./${importPath}`
+}
+
+function printGenStatus(root: string) {
+  // Is empty for JavaScript users
+  if (generatedShields.length === 0) return
+  printIntroForVite()
+  printSuccesses(root)
+  printFailures(root)
+}
+
+function printIntroForVite() {
+  console.log(`${pc.cyan(`telefunc v${projectInfo.projectVersion}`)} ${pc.green('shield() generation')}`)
+}
+
+function printFailures(root: string) {
+  const failures = generatedShields.filter((s) => s.failed)
+
+  const projects = unique(failures.map((f) => f.project))
+
+  let hasTypeScriptErrors = false
+  projects.forEach((project) => {
+    const diagnostics = project.getPreEmitDiagnostics()
+    if (diagnostics.length > 0) {
+      hasTypeScriptErrors = true
+      if (project.tsConfigFilePath) {
+        console.log(`TypeScript project ${pc.bold(project.tsConfigFilePath)} errors:`)
+      } else {
+        console.log(`TypeScript errors:`)
+      }
+      console.log(project.formatDiagnosticsWithColorAndContext(diagnostics))
+    }
+  })
+
+  const docsLink = 'https://telefunc.com/shield#typescript'
+  assertWarning(
+    failures.length === 0,
+    [
+      'Failed to generate shield() for telefunction',
+      failures.length === 1 ? '' : 's',
+      ' ',
+      formatGeneratedShields(failures, root),
+      '. ',
+      hasTypeScriptErrors
+        ? `TypeScript errors (printed above) can be problematic for shield() generation. Fix your TypeScript errors and try again. See ${docsLink} for more information.`
+        : `See ${docsLink} for how to fix this.`
+    ].join(''),
+    { onlyOnce: true }
+  )
+}
+
+function printSuccesses(root: string) {
+  const successes = generatedShields.filter((s) => !s.failed)
+  const hasFailures = successes.length < generatedShields.length
+  if (successes.length > 0) {
+    console.log(
+      [
+        pc.green('✓'),
+        `${successes.length} shield() generated`,
+        (() => {
+          const onlyOne = successes.length === 1
+          if (!hasFailures) {
+            return `covering ${onlyOne ? 'the telefunction' : 'all telefunctions'}:`
+          } else {
+            return `for the telefunction${onlyOne ? '' : 's'}:`
+          }
+        })(),
+        formatGeneratedShields(successes, root)
+      ].join(' ')
+    )
+  }
+}
+
+function formatGeneratedShields(generatedShields: GeneratedShield[], root: string) {
```

Comment:
Too long?

Reply:
I was thinking it's fine to print `GenerateShield[]`to the user after the build has finished, because that's what Vite does as well: upon`$ vite build` Vite shows all generated `dist/` files. So it aligns well with Vite. But yes I agree that for other stacks like Next.js we may not want to do that.

We can also just print something like `shield()` generated for 14 telefunctions. Maybe with some kind of threshold e.g. if more than 7 then print a more compact log.